src/main/java/software/amazon/cloudformation/resource/Serializer.java

```diff
@@ -76,10 +88,48 @@
         return OBJECT_MAPPER.writeValueAsString(modelObject);
     }
 
+    /*
+     * public <T> String compress(final String modelInput) throws IOException {
+     * final Map<String, String> map = new HashMap<>(); ByteArrayOutputStream
+     * byteArrayOutputStream = new ByteArrayOutputStream(); GZIPOutputStream gzip =
+     * new GZIPOutputStream(byteArrayOutputStream);
+     * gzip.write(modelInput.getBytes(StandardCharsets.UTF_8)); gzip.close();
+     * map.put(COMPRESSED,
+     * Base64.encodeBase64String(byteArrayOutputStream.toByteArray()));
+     * map.put(COMPRESSION_METHOD, "gzip_base64"); return
+     * OBJECT_MAPPER.writeValueAsString(map); }
+     */
+
+    public <T> String compress(final String modelInput) throws IOException {
+        final Map<String, String> map = new HashMap<>();
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
```

Comment:
I think you can do with one try

```
try (
ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
GZIPOutputStream gzip = new GZIPOutputStream(byteArrayOutputStream);
) {
...
}
```

Reply:
Seems the two `try` is right as it needs to finish first gzip output stream then starts the second one.